osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs

```diff
@@ -68,96 +62,53 @@ public void ApplyGlobal(SoloScoreInfo score, MySqlConnection conn)
-
-            if (beatmapStore == null || buildStore == null || currentTimestamp - lastStoreRefresh > 60)
-            {
-                beatmapStore = await BeatmapStore.CreateAsync(connection, transaction);
-                buildStore = await BuildStore.CreateAsync(connection, transaction);
-                lastStoreRefresh = currentTimestamp;
-            }
 
             List<SoloScore> scores = (await connection.QueryAsync<SoloScore>(
-                "SELECT * FROM scores WHERE `user_id` = @UserId AND `ruleset_id` = @RulesetId AND `preserve` = 1", new
+                "SELECT beatmap_id, pp, accuracy FROM scores s "
+                + "JOIN osu_beatmaps USING (beatmap_id) WHERE "
+                // TODO: This adds a small amount of overhead to the query.
+                // Replace this with a cached lookup once we have a good system in place.
+                + "`approved` in (1,2) AND "
+                + "s.`user_id` = @UserId AND "
+                + "`ruleset_id` = @RulesetId AND"
+                + " `pp` IS NOT NULL AND "
+                + "`preserve` = 1 AND "
+                + "`passed` = 1 AND "
+                + "`ranked` = 1", new
                 {
                     UserId = userStats.user_id,
                     RulesetId = rulesetId
                 }, transaction: transaction)).ToList();
 
-            Dictionary<uint, Beatmap?> beatmaps = new Dictionary<uint, Beatmap?>();
-
-            foreach (var score in scores)
-            {
-                if (beatmaps.ContainsKey(score.beatmap_id))
-                    continue;
-
-                beatmaps[score.beatmap_id] = await beatmapStore.GetBeatmapAsync(score.beatmap_id, connection, transaction);
-            }
-
-            // Filter out invalid scores.
-            scores.RemoveAll(s =>
-            {
-                // Score must have a valid pp.
-                if (s.pp == null)
-                    return true;
-
-                // Score must be a pass (safeguard - should be redundant with preserve flag).
-                if (!s.passed)
-                    return true;
-
-                // Beatmap must exist.
-                if (!beatmaps.TryGetValue(s.beatmap_id, out var beatmap) || beatmap == null)
-                    return true;
-
-                // Given beatmap needs to be allowed to give performance.
-                if (!beatmapStore.IsBeatmapValidForPerformance(beatmap, s.ruleset_id))
-                    return true;
-
-                // Legacy scores are always valid.
-                if (s.legacy_score_id != null)
-                    return false;
-
-                // Some older lazer scores don't have build IDs.
-                if (s.build_id == null)
-                    return true;
-
-                // Performance needs to be allowed for the build.
-                if (buildStore.GetBuild(s.build_id.Value)?.allow_performance != true)
-                    return true;
```

Comment:
I don't see any of these conditions reflected in the sql query, is it safe to assume `ranked` flag will cover that?

They don't seem to be necessarily very relevant right now (legacy scores are still processed by `osu-performance` and imported directly, old lazer scores don't exist in new tables) but yeah.

Reply:
yep

Reply:
This has been added to the score performance processor now.